### PR TITLE
Fix whisper

### PR DESCRIPTION
**Pull request by def-**
## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by heinrich5991** [comment]
Could you remove the revert? I updated the antibot on the servers to no longer crash. If you revert it again, the antibot will crash again. The Windows crash is unrelated to our server crash.

File: src/game/server/gamecontext.cpp [hunk 1]
```diff
@@ -4673,8 +4673,13 @@ void CGameContext::WhisperId(int ClientId, int VictimId, const char *pMessage)
 	if(!CheckClientId2(VictimId))
 		return;
 
-	if(m_apPlayers[ClientId])
-		m_apPlayers[ClientId]->m_LastWhisperTo = VictimId;
+	if(!m_apPlayers[ClientId])
+		return;
+
+	if(!m_apPlayers[VictimId])
```
Comment (heinrich5991) [inline src/game/server/gamecontext.cpp:4679]: Why is this function getting passed an invalid client ID? The function calling this function should probably sanitize it instead.

The proposed changes simply hide the programming error in the calling function.

It seems the relevant calling function is `CGameContext::Converse`.

But why is its `m_LastWhisperTo` invalid?

Comment (def-) [reply src/game/server/gamecontext.cpp:4679]: player left probably

Comment (heinrich5991) [reply src/game/server/gamecontext.cpp:4679]: Why does the relevant removal in

https://github.com/ddnet/ddnet/blob/95d6d99affae4787eee3f4596a90a2077504bdae/src/game/server/gamecontext.cpp#L1734-L1739

not work?

Does that mean my whispers can go to the wrong person?

Comment (KebsCS) [reply src/game/server/gamecontext.cpp:4679]: Same problem as I had in #9083, `OnClientDrop()` is only called when client is fully connected and then disconnects, if he disconnects during map reload, during reconnecting etc. the ids aren't cleared

Comment (heinrich5991) [reply src/game/server/gamecontext.cpp:4679]: If the client never really connected, why does my `m_LastWhisperTo` point to that person?

We're apparently looking at something we don't completely understand, that's why I don't like the "if something went wrong, simply ignore the request" at the lowest level. It doesn't fix the actual problem.

Comment (KebsCS) [reply src/game/server/gamecontext.cpp:4679]: He could have been connected, whispered someone and then changed his state, for example, by pressing 'Abort' during a map reload

**Comment by def-** [comment]
Please don't switch away from this branch on our prod. servers. This is crashing all the time and I can't get it merged unfortunately.

**Comment by def-** [comment]
This is important to merge, whispers currently cause crashes.

**Comment by Robyt3** [comment]
> This is important to merge, whispers currently cause crashes.

I tried to reproduce this before but I couldn't find anything even with additional assertions. I checked again and it looks like it crashes specifically with 0.7 whispers. Do you have a stack track to confirm that this is also what's crashing our servers?

**Comment by def-** [comment]
I think I deleted them because of disk space.

**Pull request closed without merging**